rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/ProjectController.java

```diff
@@ -2176,4 +2191,87 @@ private void updateReleaseNodeData(ReleaseNode releaseNode, ProjectOperation ope
             }
         }
     }
+
+    private ClearingRequest convertToClearingRequest(Map<String, Object> requestBody, String projectId) {
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        ClearingRequest clearingRequest = mapper.convertValue(requestBody, ClearingRequest.class);
+        return clearingRequest;
+    }
+
+    /**
+     * Creates a clearing request for a project.
+     *
+     * @param id          The project ID.
+     * @param reqBodyMap  The clearing request
+     * @return            The response entity containing the result of the operation.
+     * @throws TException If an error occurs during the operation.
+     */
+    @PreAuthorize("hasAuthority('WRITE')")
+    @Operation(
+            summary = "Create a clearing request for a project.",
+            tags = {"Projects"}
+    )
+    @RequestMapping(value = PROJECTS_URL + "/{id}/clearingRequest", method = RequestMethod.POST)
+    public ResponseEntity<?> createClearingRequest(
+            @Parameter(description = "Project ID", example = "376576")
+            @PathVariable("id") String id,
+            @Parameter(description = "Clearing request",
+                    schema = @Schema(implementation = ClearingRequest.class))
+            @RequestBody Map<String, Object> reqBodyMap
+    ) throws TException {
+        User sw360User = restControllerHelper.getSw360UserFromAuthentication();
+        Project sw360Project = projectService.getProjectForUserById(id, sw360User);
+        ClearingRequest clearingRequest = convertToClearingRequest(reqBodyMap, id);
+        clearingRequest.setProjectId(id);
+        clearingRequest.setRequestingUser(sw360User.getEmail());
+        clearingRequest.setClearingState(ClearingRequestState.NEW);
+
+        if (clearingRequest.getRequestingUserComment() == null) {
+            clearingRequest.setRequestingUserComment("");
+        }
+        if (clearingRequest.getClearingType() == null) {
+            return new ResponseEntity<String>("clearingType is a mandatory field. Possible values are:"
+                    + Arrays.asList(ClearingRequestType.values()), HttpStatus.BAD_REQUEST);
+        }
+        //Generated by AI(GithubCopilot)
+        if (clearingRequest.getRequestedClearingDate() != null) {
```

Comment:
`RequestedClearingDate` limit is configurable based on property via [CUSTOM_FIELD_PREFERRED_CLEARING_DATE_LIMIT ](https://github.com/eclipse-sw360/sw360/blob/main/frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/common/PortalConstants.java#L435) 

Please refer the following code: [ProjectPortlet](https://github.com/eclipse-sw360/sw360/blob/main/frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/projects/ProjectPortlet.java#L563)

Reply:
I cannot access the value of CUSTOM_FIELD_PREFERRED_CLEARING_DATE_LIMIT as PortletRequest and PortletResponse is not accessible via rest api.

Reply:
I agree, because `CUSTOM_FIELDS` are part of Liferay framework.
I suggest at least add the configuration in properties file. So that once Liferay is out, it would still be possible to configure the `RequestedClearingDate` selection.

Reply:
Currently the user can give any future or current date for the requestedClearingDate field in the request body. I agree that the configured PreferredClearingDateLimit has to be considered while setting this field . But this issue of making the Liferay Custom fields available in SW360 Properties has to be taken up separately.